File: lib/layer2_core/dark/e_secondary_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ESecondaryDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd25;
      case 'standard': return cstd25;
      case 'blue': return cbd25;
      case 'green': return cgd25;
      case 'red': return crd25;
      case 'nothing': return cnod25;
      default: return cstd25;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ESecondaryDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESecondaryDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd25
            AppTheme.STANDARD -> R.color.cstd25
            AppTheme.BLUE -> R.color.cbd25
            AppTheme.GREEN -> R.color.cgd25
            AppTheme.RED -> R.color.crd25
            AppTheme.NOTHING -> R.color.cnod25
        }
    }
}
